### ISSUE TITLE
1.13.2 Command Mission type Blocks command from being performed.

### ISSUE THREAD

**Issue by RustyDagger** [ezeiger92/QuestWorld-Extensions#5]
I tried making a command type mission and set the command to be /rankup from the plugin CMI
When I got to this quest I typed /rankup and the quest completed ( Great ) however CMI seemed to be completely unaware that its command had been used and did nothing so kind of defeats the purpose of the quest.

Also it would be great if there was some documentation explaining how to use the following mission types.

Statistic:  How do i pick the stat and the value?
Do Quest: How do i set the quest it just gives a blank map with no instructions?

**Comment by ezeiger92** [comment]
When a command mission detects a command, it is consumed and not run on the server. There are 2 ways around this:
1. Assuming it is the only mission in a quest (or the last mission in a quest), you can use `/rankup` in the command rewards
2. Add `/rankup` to the dialogue of that mission, which will be run as soon as the mission is completed

The buttons for the statistic and 'do quest' missions seem to have never been implemented. Whoops. lol

Here's how they will work after I actually make them work:
Statistic will have an amount button and a button to set the statistic as a string (you can find the list of statistics [here](https://hub.spigotmc.org/javadocs/spigot/index.html?overview-summary.html))
For entity and block based statistics, an extra button will be visible to set the entity/block.

The 'do quest' button will open a quest selector like that for picking requirements

**Comment by ezeiger92** [comment]
In typical fashion, while fixing this I made changes to QuestWorld that are required for this extension to function. I also put barely tested new features into a release candidate build, because that's totally what you should do with a release candidate. -_-

tl;dr I should have a build of QuestWorld and a patch for Extras up sometime tomorrow.

**Comment by RustyDagger** [comment]
I tried option 2 for fixing the rankup command.
It seemed to just make the player type the command twice and still did not let CMI process the command. 
Meh maybe ill give setting it at the quest reward a go? That will need to be run like a console command yes? is there a placeholder for the players name?

RIP it would seem CMI does not allow me to make players rankup from console only from ingame Weird...

**Comment by RustyDagger** [comment]
Also on the note of fresh code in the RC im sure it will be fine, almost every thing else has worked with out any flaws mostly just dumb user error on my part. If not ill be right here to help you fix it up 😄 

iv been running servers a long long time, im pretty good at debugging issues by now and that usually leads to better bug reports. ( when im not being lazy as all hell. )
